app/src/main/java/org/lineageos/glimpse/repository/MediaFlow.kt

```diff
@@ -0,0 +1,126 @@
+            sortOrder,
+            bundleOf().apply {
+                // Exclude trashed media unless we want data for the trashed album
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                    when (bucketId) {
+                        MediaStoreBuckets.MEDIA_STORE_BUCKET_TRASH.id -> {
+                            putInt(MediaStore.QUERY_ARG_MATCH_TRASHED, MediaStore.MATCH_ONLY)
+                        }
+
+                        else -> {
+                            putInt(MediaStore.QUERY_ARG_MATCH_TRASHED, MediaStore.MATCH_EXCLUDE)
+                        }
+                    }
+                }
+            })
+
+        val onLoadCompleteListener = OnLoadCompleteListener<Cursor> { _, data: Cursor? ->
+            if (!isActive) return@OnLoadCompleteListener
+            launch(Dispatchers.IO) {
+                val media = mutableListOf<Media>()
+                data?.let {
+                    val idIndex = it.getColumnIndex(MediaStore.Files.FileColumns._ID)
+                    val displayNameIndex =
+                        it.getColumnIndex(MediaStore.Files.FileColumns.DISPLAY_NAME)
+                    val isFavoriteIndex =
+                        it.getColumnIndex(MediaStore.Files.FileColumns.IS_FAVORITE)
+                    val isTrashedIndex = it.getColumnIndex(MediaStore.Files.FileColumns.IS_TRASHED)
+                    val mediaTypeIndex = it.getColumnIndex(MediaStore.Files.FileColumns.MEDIA_TYPE)
+                    val mimeTypeIndex = it.getColumnIndex(MediaStore.Files.FileColumns.MIME_TYPE)
+                    val dateAddedIndex = it.getColumnIndex(MediaStore.Files.FileColumns.DATE_ADDED)
+                    val bucketIdIndex = it.getColumnIndex(MediaStore.MediaColumns.BUCKET_ID)
+
+                    it.moveToFirst()
+                    while (!it.isAfterLast) {
+                        val id = it.getLong(idIndex)
+                        val displayName = it.getString(displayNameIndex)
+                        val isFavorite = it.getInt(isFavoriteIndex)
+                        val isTrashed = it.getInt(isTrashedIndex)
+                        val mediaType = it.getInt(mediaTypeIndex)
+                        val mimeType = it.getString(mimeTypeIndex)
+                        val dateAdded = it.getLong(dateAddedIndex)
+                        val buckedId = it.getInt(bucketIdIndex)
+
+                        media.add(
+                            Media.fromMediaStore(
+                                id,
+                                displayName,
+                                isFavorite,
+                                isTrashed,
+                                mediaType,
+                                mimeType,
+                                dateAdded,
+                                buckedId,
+                            )
+                        )
+                        it.moveToNext()
+                    }
+                    send(media)
```

Comment:
declare media inside data?.let

Reply:
actually no, I have to move it outside of the let, I need to send a zero lenght list in case there's no items in the cursor